### DISCUSSION TITLE
API documentation

### DISCUSSION THREAD

**Discussion by smlsr** [Motion-Project/motionplus#88] [General]
Is there a location where all API endpoints, required/optional params are documented?

I'd like to add some automation as well as triggers and such - looking to see how best i can build some interfaces.

**Comment by Mr-Dave** [comment]
Not really.  

The [examples](https://motion-project.github.io/motionplus_examples.html#webcontrol_pages) show the json/image feeds.  

The [sample page](https://github.com/Motion-Project/motionplus/blob/02ed21cd1d40e301f6334785c8666f25ada9890d/doc/samplepage.html#L493) illustrates the standard web page and how it interacts.

Review of the code is also possible.  [This section](https://github.com/Motion-Project/motionplus/blob/02ed21cd1d40e301f6334785c8666f25ada9890d/src/webu.cpp#L777) shows the commands possible via "GET" .  While [this section](https://github.com/Motion-Project/motionplus/blob/02ed21cd1d40e301f6334785c8666f25ada9890d/src/webu_post.cpp#L575) shows the processing of "POST" commands

**Comment by gabbymouth** [comment]
If it's any help at all, this is how I perform a "trigger":

#!/bin/bash
curl -X POST http://localhost:8080 -d "camid=$1&command=eventstart"
sleep 10
curl -X POST http://localhost:8080 -d "camid=$1&command=eventend"
